src/System.Security.Cryptography.Cng/src/Microsoft/Win32/SafeHandles/NCryptSafeHandles.cs

```diff
@@ -194,17 +211,18 @@ private enum OwnershipState {
         ///     See code:Microsoft.Win32.SafeHandles.SafeNCryptHandle#NCryptHandleDuplicationAlgorithm for
         ///     details about the algorithm.
         /// </summary>
-        private T DuplicateDuplicatedHandle<T>() where T : SafeNCryptHandle, new() {
-            Contract.Ensures(m_ownershipState == OwnershipState.Duplicate);
+        private T DuplicateDuplicatedHandle<T>() where T : SafeNCryptHandle, new()
```

Comment:
I'm trying to understand why it's ok that we're addref'ing and suppressing finalization (seems like that's a recipe for leaking these handles), so I want to look at the algorithm mentioned in the comment, but I can't find NCryptHandleDuplicationAlgorithm... where is that?